src/chainparams.cpp

```diff
@@ -349,11 +339,22 @@ class CRegTestParams : public CChainParams {
         consensus.vDeployments[d].nStartTime = nStartTime;
         consensus.vDeployments[d].nTimeout = nTimeout;
     }
-    void UpdateVersionBitsParametersFromArgs(const ArgsManager& args);
+    void UpdateActivationParametersFromArgs(const ArgsManager& args);
 };
 
-void CRegTestParams::UpdateVersionBitsParametersFromArgs(const ArgsManager& args)
+void CRegTestParams::UpdateActivationParametersFromArgs(const ArgsManager& args)
 {
+    if (gArgs.IsArgSet("-segwitheight")) {
```

Comment:
It would be nicer to do something more generic that works for other deployments, not just segwit. For example, if anybody was relying on vparams to test csv activation, it won't be possible after this PR.
But perhaps this is out of the scope for this PR, 
With something like https://github.com/bitcoin/bitcoin/pull/8994 we could simply make the all the buried heights configurable for custom regtests.

Reply:
Perhaps we cna call the parameter "-con_segwitheight: like in #8994 ?

Reply:
This argument is just for testing, so it's not disruptive to change this in future if we think the `-con_xxx` naming scheme is better. For now, I'd like to keep this as it is, rather than optimizing for a PR that hasn't been merged yet.